src/main/docker/Dockerfile

```diff
@@ -42,35 +42,36 @@ RUN zypper -n refresh && \
     zypper -n clean --all
 
 #
-# Stage 2: Download and verify gosu
+# Stage 2: Download and compile su-exec
 #
 FROM stage1 AS stage2
 
-# Install dirmngr to enable gosu verification
-RUN zypper -n install dirmngr
-
-# Download and verify gosu
-RUN gpg --batch --keyserver-options http-proxy=${env.HTTP_PROXY} --keyserver hkps://keys.openpgp.org \
-        --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4 && \
-    curl -o /usr/local/bin/gosu -SL "https://github.com/tianon/gosu/releases/download/1.17/gosu-amd64" && \
-    curl -o /usr/local/bin/gosu.asc -SL "https://github.com/tianon/gosu/releases/download/1.17/gosu-amd64.asc" && \
-    gpg --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu && \
-    chmod +x /usr/local/bin/gosu
+# Download and compile su-exec
+RUN zypper -n refresh && \
+    zypper -n update && \
+    zypper -n install gcc && \
+    curl -o /usr/local/bin/su-exec.c -SL "https://raw.githubusercontent.com/ncopa/su-exec/master/su-exec.c" && \
```

Comment:
With a direct download from the internet like this we need to verify that what has been downloaded is what we expect, i.e. by checking it's digest.

Here is an example where we are using `sha512sum -c` to do this:
https://github.com/CAFapi/buildenv-jdk11-image/blob/v1.6.3/src/main/docker/Dockerfile#L44-L45